### PR TITLE
Feature/exe 1732 update to pixelator 0.17.1

### PR DESCRIPTION
**Pull request by johandahlberg**
Updates the pipeline to use pixelator 0.17.1 and add a module for the `pixelator single-cell layout`  command.

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/pixelator/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/pixelator _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by fbdtemme — COMMENTED** [review]
Looks good! Just one thing missing:
We need to pass the layout report_json and metadata channels from PIXELATOR_LAYOUT to GENERATE_REPORTS as well similar to how we do it for all the other stages.
Right now the layout step will be missing from the qc report.

**Comment by johandahlberg** [comment]
Super. I have pushed a fix now.

File: workflows/pixelator.nf [hunk 1]
```diff
@@ -223,6 +231,7 @@ workflow PIXELATOR {
     ch_cluster_data     = PIXELATOR_GRAPH.out.all_results
     ch_annotate_data    = PIXELATOR_ANNOTATE.out.all_results
     ch_analysis_data    = PIXELATOR_ANALYSIS.out.all_results
+    ch_layout_data      = PIXELATOR_ANALYSIS.out.all_results
```
[the remark this answers is not in the document]
Comment (johandahlberg) [reply workflows/pixelator.nf:234]: Thanks!

**Review by fbdtemme — APPROVED** [review]
Ready to go!

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,26 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [[1.2.0](https://github.com/nf-core/pixelator/releases/tag/1.1.0)] - 2024-05-??
```
Comment (ambarrio) [inline CHANGELOG.md:6]: I think the link is outdated on this, right?

File: docs/output.md [hunk 3]
```diff
@@ -168,7 +168,7 @@ newly recovered components are stored in a file (components_recovered.csv).
 - `pixelator`
 
   - `annotate`
-    - `<sample-id>.dataset.pxl`
+    - `<sample-id>.annotated.dataset.pxl`
```
Comment (ambarrio) [inline docs/output.md:171]: annotated -> annotate

Let's use the stage name instead of a participle.

File: docs/output.md [hunk 4]
```diff
@@ -168,7 +168,7 @@ newly recovered components are stored in a file (components_recovered.csv).
...
     - `<sample-id>.meta.json`: Command invocation metadata.
     - `<sample-id>.rank_vs_size.png`
```
Comment (ambarrio) [inline docs/output.md:173]: These PNG files are not produced AFAIK

File: docs/output.md [hunk 5]
```diff
@@ -168,7 +168,7 @@ newly recovered components are stored in a file (components_recovered.csv).
...
     - `<sample-id>.raw_components_metrics.csv`
```
Comment (ambarrio) [inline docs/output.md:174]: These are csv.gz instead

File: CHANGELOG.md (hunk 2 above)
Comment (johandahlberg) [reply CHANGELOG.md:6]: Yes, good catch!

File: docs/output.md (hunk 3 above)
Comment (johandahlberg) [reply docs/output.md:171]: Your right, it should be annotate.

File: docs/output.md (hunk 4 above)
Comment (johandahlberg) [reply docs/output.md:173]: Also true!

File: docs/output.md (hunk 5 above)
Comment (johandahlberg) [reply docs/output.md:174]: Good catch! I will update all this things in an incoming PR.

**Pull request merged**